internal/expr/prepare.go

```diff
@@ -60,22 +60,27 @@ func starCount(fns []fullName) int {
 	return s
 }
 
+type typeNameToInfo map[string]typeInfo
+
+func typeNotPassedError(typeName string, allTypeNames []string) error {
+	if len(allTypeNames) == 0 {
+		return fmt.Errorf(`type %q not passed as a parameter`, typeName)
```

Comment:
Since we're here, should we change this message to something slightly nicer? Maybe `parameter with type %q missing, have: ...`?

Reply:
Sounds good, changed to that.